docs/rules/no-danger-with-children.md

```diff
@@ -1,4 +1,6 @@
-# Prevent problem with children and props.dangerouslySetInnerHTML (react/no-danger-with-children)
+# Disallow when a DOM element is using both children and dangerouslySetInnerHTML (react/no-danger-with-children)
```

Comment:
```suggestion
# Disallow DOM elements using both children and dangerouslySetInnerHTML (react/no-danger-with-children)
```

Reply:
Sounds good but let's make any tweaks to rule descriptions in a separate PR.